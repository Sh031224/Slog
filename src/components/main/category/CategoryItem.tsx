import Link from "next/link";
import React, { memo } from "react";
import styled from "styled-components";

interface ICategoryItemProps {
  href: string;
  params: number | string;
  name: string;
  postCount?: number;
  getIsActive: (idx: number | string) => boolean;
}

const CategoryItem: React.FC<ICategoryItemProps> = ({
  href,
  params,
  name,
  postCount,
  getIsActive
}) => {
  return (
    <Link href={href}>
      <a>
        <CategoryItemTitle isActive={getIsActive(params)}>
          {name}
          {postCount !== undefined && <CategoryPostCount>{` (${postCount})`}</CategoryPostCount>}
        </CategoryItemTitle>
      </a>
    </Link>
  );
};

const CategoryItemTitle = styled.div<{ isActive: boolean }>`
  color: ${({ theme, isActive }) => (isActive ? theme.colors.ftBlue : theme.colors.ftGray)};
  font-weight: ${({ isActive }) => (isActive ? 700 : 400)};
  margin-top: 0.3rem;
  font-size: 0.9rem;
  cursor: pointer;
`;

const CategoryPostCount = styled.span`
  color: ${({ theme }) => theme.colors.ftLightGray};
`;

export default memo(CategoryItem);
