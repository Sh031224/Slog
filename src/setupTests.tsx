import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import { lightTheme } from "style/theme";
import { store } from "stores";
import { Provider } from "react-redux";
import { NotificationContainer } from "react-notifications";

const Providers = ({ children }) => {
  return (
    <Provider store={store}>
      <ThemeProvider theme={lightTheme}>
        <NotificationContainer />
        {children}
      </ThemeProvider>
    </Provider>
  );
};

const customRender = (ui, options = {}) => render(ui, { wrapper: Providers, ...options });

// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

(window as any).scrollTo = () => {};

export { customRender as render };
