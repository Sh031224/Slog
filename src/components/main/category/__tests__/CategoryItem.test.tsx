import { render } from "setupTests";
import CategoryItem, { ICategoryItemProps } from "../CategoryItem";

describe("Category", () => {
  const setup = (props: Partial<ICategoryItemProps> = {}) => {
    const initialProps: ICategoryItemProps = {
      href: "/",
      params: "",
      name: "Testing",
      postCount: 99,
      getIsActive: () => true
    };
    return render(<CategoryItem {...initialProps} {...props} />);
  };

  it("renders correctly", () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it("item render", () => {
    const { getByText } = setup();

    getByText("Testing");
  });
});
