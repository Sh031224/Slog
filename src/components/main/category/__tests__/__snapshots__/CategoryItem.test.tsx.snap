// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category renders correctly 1`] = `
.c0 {
  color: #5373f8;
  font-weight: 700;
  margin-top: 0.3rem;
  font-size: 0.9rem;
  cursor: pointer;
}

.c1 {
  color: #bdbdbd;
}

<div>
  <div
    class="notification-container notification-container-empty"
  >
    <div />
  </div>
  <a
    href="/"
  >
    <div
      class="c0"
    >
      Testing
      <span
        class="c1"
      >
         (99)
      </span>
    </div>
  </a>
</div>
`;
